android/demos/compose/src/main/kotlin/com/uber/rib/compose/root/main/logged_in/LoggedInInteractor.kt

```diff
@@ -38,10 +40,14 @@ class LoggedInInteractor(
 
   override fun didBecomeActive(savedInstanceState: Bundle?) {
     super.didBecomeActive(savedInstanceState)
+
     eventStream.observe()
-      .ofType<LoggedInEvent.LogOutClick>()
-      .autoDispose(this)
-      .subscribe { authStream.accept(AuthInfo(false)) }
+      .onEach {
```

Comment:
Why not use the `filterIsInstance` operator instead of checking the instance in `onEach`?

```kotlin
eventStream.observe()
    .filterIsInstance<LoggedInEvent.LogOutClick>()
    .onEach { authStream.accept(AuthInfo(false)) }
    .launchIn(coroutineScope)
```

Reply:
The original approach had this, however with the when() on sealed classes we can compiler guarantee all use-cases are covered so would be preferred.

Reply:
Understood -- in cases where multiple children of sealed classes are dealt with, we'd just add new cases to `when`. You can go ahead and resolve all comments related to `filterIsInstance`.